src/main/scala/Column.scala

```diff
@@ -1,22 +1,294 @@
+    .objectNode()
+    .set("@id", new TextNode("http://www.w3.org/2001/XMLSchema#string"))
+
+  def getOrdered(inheritedProperties: Map[String, JsonNode]): Boolean = {
+    val inheritedPropertiesNode = inheritedProperties.get("ordered")
+    inheritedPropertiesNode match {
+      case Some(value) => value.asBoolean()
+      case _           => false
+    }
+  }
+
+  def getTextDirection(inheritedProperties: Map[String, JsonNode]): String = {
+    val textDirectionNode = inheritedProperties.get("textDirection")
+    textDirectionNode match {
+      case Some(value) => value.asText()
+      case _           => "inherit"
+    }
+  }
+
+  def getSuppressOutput(columnProperties: Map[String, JsonNode]): Boolean = {
+    val suppressOutputNode = columnProperties.get("suppressOutput")
+    suppressOutputNode match {
+      case Some(value) => value.asBoolean()
+      case _           => false
+    }
+  }
+
+  def getVirtual(columnProperties: Map[String, JsonNode]): Boolean = {
+    val virtualNode = columnProperties.get("virtual")
+    virtualNode match {
+      case Some(value) => value.asBoolean()
+      case _           => false
+    }
+  }
+
+  def getRequired(inheritedProperties: Map[String, JsonNode]): Boolean = {
+    inheritedProperties.get("required") match {
+      case Some(value) => value.asBoolean()
+      case _           => false
+    }
+  }
+
+  def getDefault(inheritedProperties: Map[String, JsonNode]): String = {
+    inheritedProperties.get("default") match {
+      case Some(value) => value.asInstanceOf[TextNode].asText()
+      case _           => ""
+    }
+  }
+
+  def getId(columnProperties: Map[String, JsonNode]): Option[String] = {
+    val idNode = columnProperties.get("@id")
+    if (idNode.isDefined) Some(idNode.get.asText()) else None
+  }
+
+  def getName(
+      columnProperties: Map[String, JsonNode],
+      lang: String
+  ): Option[String] = {
+    val name = columnProperties.get("name")
+    val titles = columnProperties.get("titles")
+
+    if (name.isDefined) {
+      Some(name.get.asInstanceOf[TextNode].asText())
+    } else if (titles.isDefined && titles.get.path(lang).isMissingNode) {
+      val langArray = Array.from(
+        titles.get.path(lang).elements().asScala
+      )
+      if (langArray.nonEmpty) {
+        Some(langArray(0).asText())
+      } else None
+    } else None // Not sure what to return here. Hope it does not reach here
+  }
+
+  def getNullParam(
+      inheritedProperties: Map[String, JsonNode]
+  ): Array[String] = {
+    inheritedProperties.get("null") match {
+      case Some(value) => {
+        value match {
+          case a: ArrayNode => {
+            var nullParamsToReturn = Array[String]()
+            val nullParams = Array.from(a.elements.asScala)
+            for (np <- nullParams)
+              nullParamsToReturn :+= np.asText()
+            nullParamsToReturn
+          }
+          case s: TextNode => Array[String](s.asText())
+          case _ =>
```

Comment:
What happened to the case where s == null? That's a valid option which should default to "".

Reply:
That code is in PropertyChecker class, nullProperty method

Reply:
This is dealing with the null property too isn't it? So shouldn't the code be equivalent?

Reply:
Inherited properties here (`inheritedProperties: Map[String, JsonNode]`) is already processed once through the PropertyChecker class. Thus it will always be one of TextNode or ArrayNode.

Reply:
I can remove the line which raises a metadata exception here (as per current implementation it will never reach that else part of the match)

Reply:
No, that's fine.